### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by albi3ro**
This PR just has some minor documentation fixes, such as cleaning up type hints.

### CODE REVIEW

**Review by dwierichs — APPROVED** [review]
lgtm, thanks @albi3ro  :tada

**Review by AlbertMitjans — APPROVED** [review]
Thanks!

File: pennylane/pulse/parametrized_hamiltonian.py [hunk 1]
```diff
@@ -278,7 +278,7 @@ def ops(self):
         """Return the operators defining the ``ParametrizedHamiltonian``.
 
         Returns:
-            Iterable[Observable]: observables in the Hamiltonian expression
+            Iterable[Operator]): observables in the Hamiltonian expression
```
Comment (timmysilv) [inline pennylane/pulse/parametrized_hamiltonian.py:281]:
```suggestion
            Iterable[Operator]: observables in the Hamiltonian expression
```

**Review by timmysilv — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]
Thanks for the changes!

**Pull request merged**